### DISCUSSION TITLE
quire pdf error on 1.0.0-rc.6 on Ubuntu caused by missing packages

### DISCUSSION THREAD

**Discussion by chrisdaaz** [thegetty/quire#707] [Feedback]
I ran into an error building a PDF with `quire 1.0.0-rc.6` on Ubuntu 20.04 via Windows Subsystem for Linux on a Windows 11 laptop. This was on a fresh project, after first running `quire new demo-rc-6`, `quire preview`, `quire build`, and `quire epub` without problems.

Here's the error log: 
```
$ quire pdf --debug
[CLI] Command 'pdf' called with options { lib: 'pagedjs', debug: true }
[CLI:lib/pdf] generating PDF using Paged.js
[CLI:lib/pdf/pagedjs] Printer options
{
  "allowLocal": true,
  "debug": true,
  "enableWarnings": true
}
[CLI:lib/pdf/pagedjs] PDF options
{
  "outlineTags": [
    "h1"
  ]
}
[CLI:lib/pdf/pagedjs] printing /home/chris/quire/demo-rc-6/_site/pdf.html
Error: Failed to launch the browser process!
/home/chris/.cache/puppeteer/chrome/linux-1108766/chrome-linux/chrome: error while loading shared libraries: libcups.so.2: cannot open shared object file: No such file or directory


TROUBLESHOOTING: https://pptr.dev/troubleshooting

    at Interface.onClose (file:///home/chris/.nvm/versions/node/v18.13.0/lib/node_modules/@thegetty/quire-cli/node_modules/@puppeteer/browsers/lib/esm/launch.js:253:24)
    at Interface.emit (node:events:525:35)
    at Interface.close (node:internal/readline/interface:533:10)
    at Socket.onend (node:internal/readline/interface:259:10)
    at Socket.emit (node:events:525:35)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
```

I did some searching and [found something that worked](https://github.com/electron/electron-quick-start/issues/486#issuecomment-1153535808). `quire pdf` worked after I installed these packages: 

```
sudo apt install libnss3-dev libgdk-pixbuf2.0-dev libgtk-3-dev libxss-dev
```

**Comment by Erin-Cecele** [comment]
@chrisdaaz Thank you for sharing this `quire pdf` solution with the community. I'm not well-versed in Linux. Will this solution work for all Windows 11 users or just those working on Linux systems?

**Reply by chrisdaaz** [reply]
I think this might help Ubuntu users.

**Reply by Erin-Cecele** [reply]
Thank you @chrisdaaz. This is good to know. We appreciate your help!

**Comment by chrisdaaz** [comment]
I ran into this problem again and fixed `quire pdf` on Windows Subsystem for Linux (Ubuntu 20.04.3 LTS) by installing this package: `sudo apt-get install libasound2`

**Reply by Erin-Cecele** [reply]
Thank you for the update @chrisdaaz. Do you think this is something that needs to be addressed in core-Quire? Or is this simply an FYI for other Ubuntu users?
